core/src/main/java/org/elasticsearch/action/admin/cluster/shards/ClusterSearchShardsGroup.java

```diff
@@ -26,23 +26,22 @@
 import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.index.Index;
+import org.elasticsearch.index.shard.ShardId;
 
 import java.io.IOException;
 
 /**
  */
 public class ClusterSearchShardsGroup implements Streamable, ToXContent {
 
-    private Index index;
-    private int shardId;
+    private ShardId shardId;
```

Comment:
Could this be made `final`, and have a constructor that takes a `StreamInput`, and `readFrom` just delegate to the constructor?